turbo/jsonrpc/otterscan_search_trace.go

```diff
@@ -17,38 +18,31 @@ import (
 	"github.com/ledgerwatch/log/v3"
 )
 
-func (api *OtterscanAPIImpl) searchTraceBlock(ctx, traceCtx context.Context, traceCtxCancel context.CancelFunc, errCh chan<- error, addr common.Address, chainConfig *chain.Config, idx int, bNum uint64, results []*TransactionsWithReceipts) {
+func (api *OtterscanAPIImpl) searchTraceBlock(ctx context.Context, addr common.Address, chainConfig *chain.Config, idx int, bNum uint64, results []*TransactionsWithReceipts) error {
 	// Trace block for Txs
 	newdbtx, err := api.db.BeginRo(ctx)
 	if err != nil {
 		log.Error("Search trace error", "err", err)
 		results[idx] = nil
-		return
+		return nil
 	}
 	defer newdbtx.Rollback()
 
 	// found will be false when CallTraceSet info and CallFromIndex/CallToIndex table info mismatch
 	// searchTraceBlock goroutine will sequentially search every block written in CallTraceSet
 	// when mismatch, causing cpu hike. To avoid this, when inconsistency found, early terminate.
 	found, result, err := api.traceBlock(newdbtx, ctx, bNum, addr, chainConfig)
-	if !found {
+	if !found && !errors.Is(err, context.Canceled) && !errors.Is(err, context.DeadlineExceeded) {
```

Comment:
I think checking `result != nil` is better because it clearly matches the inconsistency situation.
If `found == false` and `result != nil`, error is always `nil`, due to the implementation of `api.traceBlock`. May comment this behavior.